macOS: compile failure due to htons (and the like)

current master fails on macOS with:

```
/rtrlib/rtrlib/rtr/packets.c:215:11: error: use of undeclared identifier 'htons'
                return &htons;
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,9 +23,11 @@ if(NOT APPLE)
 endif(NOT APPLE)
 include(UseMultiArch) # needed for debian packaging
 
-set(RTRLIB_SRC rtrlib/rtr_mgr.c rtrlib/lib/utils.c rtrlib/lib/ip.c rtrlib/lib/ipv4.c rtrlib/lib/ipv6.c rtrlib/lib/log.c
+set(RTRLIB_SRC rtrlib/rtr_mgr.c rtrlib/lib/utils.c rtrlib/lib/convert_byte_order.c
+    rtrlib/lib/ip.c rtrlib/lib/ipv4.c rtrlib/lib/ipv6.c rtrlib/lib/log.c
     rtrlib/pfx/trie/trie.c rtrlib/pfx/trie/trie-pfx.c rtrlib/transport/transport.c
-    rtrlib/transport/tcp/tcp_transport.c rtrlib/rtr/rtr.c rtrlib/rtr/packets.c rtrlib/spki/hashtable/ht-spkitable.c rtrlib/spki/hashtable/tommyds-1.8/tommy.c)
+    rtrlib/transport/tcp/tcp_transport.c rtrlib/rtr/rtr.c rtrlib/rtr/packets.c
+    rtrlib/spki/hashtable/ht-spkitable.c rtrlib/spki/hashtable/tommyds-1.8/tommy.c)
 set(RTRLIB_LINK ${RT_LIB} ${CMAKE_THREAD_LIBS_INIT})
 
 find_package(LibSSH 0.6.0 QUIET)

--- a/rtrlib/lib/convert_byte_order.c
+++ b/rtrlib/lib/convert_byte_order.c
@@ -1,0 +1,35 @@
+/*
+ * This file is part of RTRlib.
+ *
+ * This file is subject to the terms and conditions of the MIT license.
+ * See the file LICENSE in the top level directory for more details.
+ *
+ * Website: http://rtrlib.realmv6.org/
+ */
+
+#include <assert.h>
+#include <arpa/inet.h>
+#include <inttypes.h>
+#include "rtrlib/lib/convert_byte_order.h"
+
+uint16_t lrtr_convert_short(const enum target_byte_order tbo,
+			    const uint16_t value)
+{
+	if (tbo == TO_NETWORK_BYTE_ORDER)
+		return htons(value);
+	else if (tbo == TO_HOST_HOST_BYTE_ORDER)
+		return ntohs(value);
+
+	assert(0);
+}
+
+uint32_t lrtr_convert_long(const enum target_byte_order tbo,
+			   const uint32_t value)
+{
+	if (tbo == TO_NETWORK_BYTE_ORDER)
+		return htonl(value);
+	else if (tbo == TO_HOST_HOST_BYTE_ORDER)
+		return ntohl(value);
+
+	assert(0);
+}

--- a/rtrlib/lib/convert_byte_order.h
+++ b/rtrlib/lib/convert_byte_order.h
@@ -1,0 +1,41 @@
+/*
+ * This file is part of RTRlib.
+ *
+ * This file is subject to the terms and conditions of the MIT license.
+ * See the file LICENSE in the top level directory for more details.
+ *
+ * Website: http://rtrlib.realmv6.org/
+ */
+
+#ifndef LRTR_CONVERT_BYTE_ORDER_H
+#define LRTR_CONVERT_BYTE_ORDER_H
+
+#include <inttypes.h>
+
+/**
+ * @brief Target byte order for conversion.
+ */
+enum target_byte_order {
+	TO_NETWORK_BYTE_ORDER,
+	TO_HOST_HOST_BYTE_ORDER,
+};
+
+/**
+ * Converts the passed short value to the given target byte order.
+ * @param[in] tbo Target byte order.
+ * @param[in] value Input (uint16_t) for conversion.
+ * @result Converted uint16_t value.
+ */
+uint16_t lrtr_convert_short(const enum target_byte_order tbo,
+			    const uint16_t value);
+
+/**
+ * Converts the passed long value to the given target byte order.
+ * @param[in] tbo Target byte order.
+ * @param[in] value Input (uint32_t) for conversion.
+ * @result Converted uint32_t value.
+ */
+uint32_t lrtr_convert_long(const enum target_byte_order tbo,
+			   const uint32_t value);
+
+#endif /* LRTR_CONVERT_BYTE_ORDER_H */

--- a/rtrlib/lib/ipv4.c
+++ b/rtrlib/lib/ipv4.c
@@ -7,8 +7,9 @@
  * Website: http://rtrlib.realmv6.org/
  */
 
+#include "rtrlib/lib/convert_byte_order.h"
 #include "rtrlib/lib/ipv4.h"
-#include "rtrlib/lib/utils.h"
+
 #include <assert.h>
 #include <stdio.h>
 
@@ -46,7 +47,7 @@ bool lrtr_ipv4_addr_equal(const struct lrtr_ipv4_addr *a, const struct lrtr_ipv4
 
 void lrtr_ipv4_addr_convert_byte_order(const uint32_t src,
                                        uint32_t *dest,
-                                       uint32_t (*convert_fp)(uint32_t))
+				       const enum target_byte_order tbo)
 {
-	*dest = convert_fp(src);
+	*dest = lrtr_convert_long(tbo, src);
 }

--- a/rtrlib/lib/ipv4.h
+++ b/rtrlib/lib/ipv4.h
@@ -9,6 +9,9 @@
 
 #ifndef LRTR_IPV4_H
 #define LRTR_IPV4_H
+
+#include "rtrlib/lib/convert_byte_order.h"
+
 #include <inttypes.h>
 #include <stdbool.h>
 
@@ -58,7 +61,6 @@ int lrtr_ipv4_addr_to_str(const struct lrtr_ipv4_addr *ip, char *str, const unsi
 */
 bool lrtr_ipv4_addr_equal(const struct lrtr_ipv4_addr *a, const struct lrtr_ipv4_addr *b);
 
-
 /**
  * @ingroup util_h[{
  * Converts the passed IPv4 address byte order with the given function.
@@ -69,6 +71,6 @@ bool lrtr_ipv4_addr_equal(const struct lrtr_ipv4_addr *a, const struct lrtr_ipv4
 */
 void lrtr_ipv4_addr_convert_byte_order(const uint32_t src,
                                        uint32_t *dest,
-                                       uint32_t (*convert_fp)(uint32_t));
+				       const enum target_byte_order tbo);
 
 #endif

--- a/rtrlib/lib/ipv6.c
+++ b/rtrlib/lib/ipv6.c
@@ -7,9 +7,11 @@
  * Website: http://rtrlib.realmv6.org/
  */
 
+#include "rtrlib/lib/convert_byte_order.h"
 #include "rtrlib/lib/ipv6.h"
 #include "rtrlib/lib/ipv4.h"
 #include "rtrlib/lib/utils.h"
+
 #include <arpa/inet.h>
 #include <assert.h>
 #include <string.h>
@@ -197,8 +199,8 @@ int lrtr_ipv6_addr_to_str(const struct lrtr_ipv6_addr *ip_addr, char *b, const u
 
 void lrtr_ipv6_addr_convert_byte_order(const uint32_t *src,
                                        uint32_t *dest,
-                                       uint32_t (*convert_fp)(uint32_t))
+                                       const enum target_byte_order tbo)
 {
     for(int i = 0; i < 4; i++)
-        dest[i] = convert_fp(src[i]);
+        dest[i] = lrtr_convert_long(tbo, src[i]);
 }

--- a/rtrlib/lib/ipv6.h
+++ b/rtrlib/lib/ipv6.h
@@ -9,10 +9,12 @@
 
 #ifndef LRTR_IPV6_H
 #define LRTR_IPV6_H
+
+#include "rtrlib/lib/convert_byte_order.h"
+
 #include <sys/types.h>
 #include <stdbool.h>
 #include <stdint.h>
-
 
 /**
  * @brief Struct holding an IPv6 address in host byte order.
@@ -69,6 +71,6 @@ int lrtr_ipv6_str_to_addr(const char *a, struct lrtr_ipv6_addr *ip);
 */
 void lrtr_ipv6_addr_convert_byte_order(const uint32_t *src,
                                        uint32_t *dest,
-                                       uint32_t (*convert_fp)(uint32_t));
+				       const enum target_byte_order tbo);
 
 #endif

--- a/rtrlib/rtr/packets.c
+++ b/rtrlib/rtr/packets.c
@@ -11,6 +11,8 @@
 #include <string.h>
 #include <assert.h>
 #include <stdio.h>
+
+#include "rtrlib/lib/convert_byte_order.h"
 #include "rtrlib/transport/transport.h"
 #include "rtrlib/rtr/packets.h"
 #include "rtrlib/lib/utils.h"
@@ -19,11 +21,6 @@
 
 #define TEMPORARY_PDU_STORE_INCREMENT_VALUE 100
 #define MAX_SUPPORTED_PDU_TYPE 10
-
-enum target_byte_order {
-	TO_NETWORK_BYTE_ORDER,
-	TO_HOST_HOST_BYTE_ORDER,
-};
 
 enum pdu_error_type {
     CORRUPT_DATA = 0,
@@ -209,110 +206,91 @@ void rtr_change_socket_state(struct rtr_socket *rtr_socket, const enum rtr_socke
         rtr_socket->connection_state_fp(rtr_socket, new_state,rtr_socket->connection_state_fp_param);
 }
 
-static uint16_t (*get_convert_short_fp(enum target_byte_order target_byte_order))(uint16_t)
+static void rtr_pdu_convert_header_byte_order(void *pdu, const enum target_byte_order target_byte_order)
 {
-	if (target_byte_order == TO_NETWORK_BYTE_ORDER)
-		return &htons;
-	else if (target_byte_order == TO_HOST_HOST_BYTE_ORDER)
-		return &ntohs;
-	else
-		assert(0);
-}
-
-static uint32_t (*get_convert_long_fp(enum target_byte_order target_byte_order))(uint32_t)
-{
-	if (target_byte_order == TO_NETWORK_BYTE_ORDER)
-		return &htonl;
-	else if (target_byte_order == TO_HOST_HOST_BYTE_ORDER)
-		return &ntohl;
-	else
-		assert(0);
-}
-
-static void rtr_pdu_convert_header_byte_order(void *pdu, enum target_byte_order target_byte_order)
-{
-	uint16_t (*convert_short)(uint16_t) = get_convert_short_fp(target_byte_order);
-	uint32_t (*convert_long)(uint32_t) = get_convert_long_fp(target_byte_order);
 	struct pdu_header *header = pdu;
 
 	//The ROUTER_KEY PDU has two 1 Byte fields instead of the 2 Byte reserved field.
 	if (header->type != ROUTER_KEY)
-		header->reserved = convert_short(header->reserved);
+		header->reserved = lrtr_convert_short(target_byte_order, header->reserved);
 
-	header->len = convert_long(header->len);
+	header->len = lrtr_convert_long(target_byte_order, header->len);
 }
 
-static void rtr_pdu_convert_footer_byte_order(void *pdu, enum target_byte_order target_byte_order) {
-	uint32_t (*convert_long)(uint32_t) = get_convert_long_fp(target_byte_order);
+static void rtr_pdu_convert_footer_byte_order(void *pdu, const enum target_byte_order target_byte_order) {
 	struct pdu_error *err_pdu;
 	struct pdu_header *header = pdu;
 	uint32_t addr6[4];
 	const enum pdu_type type = rtr_get_pdu_type(pdu);
 
-
-
 	switch (type) {
 	case SERIAL_QUERY:
-		((struct pdu_serial_query *) pdu)->sn = convert_long(((struct pdu_serial_query *) pdu)->sn);
+		((struct pdu_serial_query *) pdu)->sn =
+			lrtr_convert_long(target_byte_order, ((struct pdu_serial_query *) pdu)->sn);
 		break;
 
 	case ERROR:
 		err_pdu = (struct pdu_error *) pdu;
 		if (target_byte_order == TO_NETWORK_BYTE_ORDER) {
 			*((uint32_t *)(err_pdu->rest + err_pdu->len_enc_pdu)) =
-			    convert_long(*((uint32_t *)(err_pdu->rest + err_pdu->len_enc_pdu)));
-			err_pdu->len_enc_pdu = convert_long(err_pdu->len_enc_pdu);
+			    lrtr_convert_long(target_byte_order, *((uint32_t *)(err_pdu->rest + err_pdu->len_enc_pdu)));
+			err_pdu->len_enc_pdu = lrtr_convert_long(target_byte_order, err_pdu->len_enc_pdu);
 		} else {
-			err_pdu->len_enc_pdu = convert_long(err_pdu->len_enc_pdu);
+			err_pdu->len_enc_pdu = lrtr_convert_long(target_byte_order, err_pdu->len_enc_pdu);
 			*((uint32_t *)(err_pdu->rest + err_pdu->len_enc_pdu)) =
-			    convert_long(*((uint32_t *)(err_pdu->rest + err_pdu->len_enc_pdu)));
+			    lrtr_convert_long(target_byte_order, *((uint32_t *)(err_pdu->rest + err_pdu->len_enc_pdu)));
 		}
 		break;
 
 	case SERIAL_NOTIFY:
 		((struct pdu_serial_notify *) pdu)->sn =
-		    convert_long(((struct pdu_serial_notify *) pdu)->sn);
+		    lrtr_convert_long(target_byte_order, ((struct pdu_serial_notify *) pdu)->sn);
 		break;
 
 	case EOD:
 		if (header->ver == RTR_PROTOCOL_VERSION_1) {
 			((struct pdu_end_of_data_v1 *) pdu)->expire_interval =
-			    convert_long(((struct pdu_end_of_data_v1 *) pdu)->expire_interval);
+			    lrtr_convert_long(target_byte_order,
+				    	 ((struct pdu_end_of_data_v1 *) pdu)->expire_interval);
 
 			((struct pdu_end_of_data_v1 *) pdu)->refresh_interval =
-			    convert_long(((struct pdu_end_of_data_v1 *) pdu)->refresh_interval);
+			    lrtr_convert_long(target_byte_order,
+				    	 ((struct pdu_end_of_data_v1 *) pdu)->refresh_interval);
 
 			((struct pdu_end_of_data_v1 *) pdu)->retry_interval =
-			    convert_long(((struct pdu_end_of_data_v1 *) pdu)->retry_interval);
+			    lrtr_convert_long(target_byte_order,
+				    	 ((struct pdu_end_of_data_v1 *) pdu)->retry_interval);
 
 			((struct pdu_end_of_data_v1 *) pdu)->sn =
-			    convert_long(((struct pdu_end_of_data_v1 *) pdu)->sn);
+			    lrtr_convert_long(target_byte_order,
+				    	 ((struct pdu_end_of_data_v1 *) pdu)->sn);
 		} else {
 			((struct pdu_end_of_data_v0 *) pdu)->sn =
-			    convert_long(((struct pdu_end_of_data_v0 *) pdu)->sn);
+			    lrtr_convert_long(target_byte_order,
+				    	 ((struct pdu_end_of_data_v0 *) pdu)->sn);
 		}
 		break;
 
 	case IPV4_PREFIX:
 		lrtr_ipv4_addr_convert_byte_order(((struct pdu_ipv4 *) pdu)->prefix,
 		                                  &((struct pdu_ipv4 *) pdu)->prefix,
-		                                  convert_long);
+					  	  target_byte_order);
 		((struct pdu_ipv4 *) pdu)->asn =
-		    convert_long(((struct pdu_ipv4 *) pdu)->asn);
+			lrtr_convert_long(target_byte_order, ((struct pdu_ipv4 *) pdu)->asn);
 		break;
 
 	case IPV6_PREFIX:
 		lrtr_ipv6_addr_convert_byte_order(((struct pdu_ipv6 *) pdu)->prefix,
 		                                  addr6,
-		                                  convert_long);
+		                                  target_byte_order);
 		memcpy(((struct pdu_ipv6 *) pdu)->prefix, addr6, sizeof(addr6));
 		((struct pdu_ipv6 *) pdu)->asn =
-		    convert_long(((struct pdu_ipv6 *) pdu)->asn);
+		    lrtr_convert_long(target_byte_order, ((struct pdu_ipv6 *) pdu)->asn);
 		break;
 
 	case ROUTER_KEY:
 		((struct pdu_router_key *) pdu)->asn =
-		    convert_long(((struct pdu_router_key *) pdu)->asn);
+		    lrtr_convert_long(target_byte_order, ((struct pdu_router_key *) pdu)->asn);
 		break;
 
 	default:

--- a/rtrlib/rtr/packets.c
+++ b/rtrlib/rtr/packets.c
@@ -304,7 +304,7 @@ static void rtr_pdu_header_to_network_byte_order(void *pdu)
 	rtr_pdu_convert_header_byte_order(pdu, TO_NETWORK_BYTE_ORDER);
 }
 
-static void rtr_pdu_footer_to_netwotk_byte_order(void *pdu)
+static void rtr_pdu_footer_to_network_byte_order(void *pdu)
 {
 	rtr_pdu_convert_footer_byte_order(pdu, TO_NETWORK_BYTE_ORDER);
 }
@@ -312,7 +312,7 @@ static void rtr_pdu_footer_to_netwotk_byte_order(void *pdu)
 static void rtr_pdu_to_network_byte_order(void *pdu)
 {
 	rtr_pdu_header_to_network_byte_order(pdu);
-	rtr_pdu_footer_to_netwotk_byte_order(pdu);
+	rtr_pdu_footer_to_network_byte_order(pdu);
 }
 
 static void rtr_pdu_footer_to_host_byte_order(void *pdu)

--- a/scripts/check-coding-files.txt
+++ b/scripts/check-coding-files.txt
@@ -1,5 +1,7 @@
 rtrlib/rtr_mgr.h
 rtrlib/rtr_mgr.c
+rtrlib/lib/convert_byte_order.h
+rtrlib/lib/convert_byte_order.c
 rtrlib/pfx/trie/trie.h
 rtrlib/pfx/trie/trie.c
 tests/test_ht_spkitable.c

